gips/data/prism/prism.py

```diff
@@ -117,33 +117,24 @@ def fetch_ftp(cls, asset, tile, date):
             )
             if len(filenames) > 1:
                 filenames = sorted(filenames, key=lambda x: prismAsset(x).ver_stab, reverse=True)
-            filename = filenames[0]
-            stagedir = tempfile.mkdtemp(
-                prefix='prismDownloader',
-                dir=cls.Repository.path('stage')
-            )
-            ofilename = os.path.join(stagedir, filename)
-            utils.verbose_out("Downloading %s" % filename, 2)
-            with open(ofilename, "wb") as ofile:
-                ftp.retrbinary('RETR %s' % filename, ofile.write)
-            ftp.close()
-        except Exception, e:
-            # TODO error-handling-fix: with handler BUT mind the else
-            raise Exception("Error downloading: %s" % e)
-        else:
-            assets = cls.archive(stagedir)
-        try:
-            os.remove(ofilename)
-        except OSError as ose:
-            # TODO error-handling-fix: change to 'raise'
-            if ose.errno != 2:
-                raise ose
-        os.rmdir(stagedir)
+            asset_fn = filenames[0]
+            stage_dir_fp = cls.Repository.path('stage')
+            stage_fp = os.path.join(stage_dir_fp, asset_fn)
+            with utils.make_temp_dir(prefix='fetchtmp', dir=stage_dir_fp) as td_name:
+                temp_fp = os.path.join(td_name, asset_fn)
+                utils.verbose_out("Downloading " + asset_fn, 2)
+                with open(temp_fp, "wb") as temp_fo:
+                    ftp.retrbinary('RETR ' + asset_fn, temp_fo.write)
+                ftp.quit()
+                os.rename(temp_fp, stage_fp)
+            return stage_fp
+
```

Comment:
sounds reasonable...the `fetch_ftp` was just a breadcrumb because there were only a couple little hacks to the function, which I had planned to use the base-method eventually....given that we're headed to query service, dropping `fetch_ftp` sounds reasonable.